File: examples/Experimental/Basic/2-Entity.rb
require "lumino"

class App < Application
    def on_init
        Engine.render_view.guide_grid_enabled = true
        Engine.camera.set_position(5, 5, -5)
        Engine.camera.look_at(0, 0, 0)
        
        box1 = BoxMesh.new
        box1.set_position(1, 0, 0)
        box1.add_into

        box2 = BoxMesh.new
        box2.set_position(0, 2, 0)
        box2.add_into

        box3 = BoxMesh.new
        box3.set_position(0, 0, 3)
        box3.add_into
    end
end

App.new.run

File: legacy/examples/Experimental/Basic/2-Entity.cpp
#include <Lumino.hpp>

class EntityBasicExample : public Application
{
    void onInit() override
    {
        Engine::renderView()->setGuideGridEnabled(true);
        Engine::mainCamera()->setPosition(5, 5, -5);
        Engine::mainCamera()->lookAt(0, 0, 0);

        auto box1 = BoxMesh::With().buildInto();
        box1->setPosition(1, 0, 0);

        auto box2 = BoxMesh::With().buildInto();
        box2->setPosition(0, 2, 0);

        auto box3 = BoxMesh::With().buildInto();
        box3->setPosition(0, 0, 3);
    }
};

LUMINO_APP(EntityBasicExample);
